framework/src/main/java/org/apache/drill/test/framework/TestDriver.java

```diff
@@ -555,34 +556,39 @@ public void run() {
       }
     }
   }
-  
 
-  private static void hdfsCopy(Path src, Path dest, boolean overWrite, String fsMode)
-      throws IOException {
-    LOG.debug("Copy " + src + " to " + dest);
+  private static void dfsCopy(Path src, Path dest, String fsMode)
+          throws IOException {
+
     FileSystem fs;
+    FileSystem localFs = FileSystem.getLocal(conf);
+
     if (fsMode.equals(LOCALFS)) {
       fs = FileSystem.getLocal(conf);
     } else {
       fs = FileSystem.get(conf);
     }
-    FileSystem localFs = FileSystem.getLocal(conf);
-    if (localFs.getFileStatus(src).isDir()) {
-      for (FileStatus file : localFs.listStatus(src)) {
-        Path newSrc = file.getPath();
-        Path newDest = new Path(dest, newSrc.getName());
-        hdfsCopy(file.getPath(), newDest, overWrite, fsMode);
-      }
-    } else if (!fs.exists(dest) || overWrite) {
-      try {
-        fs.copyFromLocalFile(false, overWrite, src, dest);
-      } catch (FileAlreadyExistsException e) {
-    	LOG.debug("The source file " + src
-    	          + " already exists in destination.  Skipping the copy.");
+
+    try {
+      if (localFs.getFileStatus(src).isDirectory()) {
+        for (FileStatus file : localFs.listStatus(src)) {
+          Path srcChild = file.getPath();
+          Path newDest = new Path(dest + "/" + srcChild.getName());
+          dfsCopy(srcChild, newDest, fsMode);
+        }
+      } else {
+        if (!fs.exists(dest.getParent())) {
+          fs.mkdirs(dest.getParent());
+        }
+        FileUtil.copy(localFs, src, fs, dest, false, fs.getConf());
```

Comment:
it would be nice to know the difference between FileUtil.copy and fs.copyFromLocalFile. But if all your tests prove positive, let's check it in. 

+1